elephant/spike_train_surrogates.py

```diff
@@ -46,25 +46,61 @@
 from __future__ import division, print_function, unicode_literals
 
 import random
+from functools import partial
 
 import neo
 import numpy as np
 import quantities as pq
 from scipy.ndimage import gaussian_filter
 
-try:
-    import elephant.statistics as es
-
-    isi = es.isi
-except ImportError:
-    from .statistics import isi  # Convenience when in elephant working dir.
+from elephant.statistics import isi
 
 # List of all available surrogate methods
 SURR_METHODS = ['dither_spike_train', 'dither_spikes', 'jitter_spikes',
-                'randomise_spikes', 'shuffle_isis', 'joint_isi_dithering']
+                'randomise_spikes', 'shuffle_isis', 'joint_isi_dithering',
+                'dither_spikes_with_refractory_period']
+
+
+def _dither_spikes_with_refractory_period(spiketrain, dither, n,
+                                          refractory_period):
+    units = spiketrain.units
+    t_start = spiketrain.t_start.rescale(units).magnitude
+    t_stop = spiketrain.t_stop.rescale(units).magnitude
+
+    dither = dither.rescale(units).magnitude
+    refractory_period = refractory_period.rescale(units).magnitude
+    refractory_period = np.min(np.diff(spiketrain.magnitude),
```

Comment:
maybe add a comment here?

Reply:
Isn't it already described in `dither_spikes()` docstring?

Reply:
I added another comment there to explain it also in the code.